archiver.go

```diff
@@ -9,6 +9,24 @@ import (
 	"runtime"
 )
 
+// PreservableArchiver represent a archive format
+type PreservableArchiver interface {
+	// Match checks supported files
+	Match(filename string) bool
+	// Make makes an archive file on disk.
+	Make(destination string, sources []string) error
+	// Open extracts an archive file on disk.
+	Open(source, destination string) error
+	// Write writes an archive to a Writer.
+	Write(output io.Writer, sources []string) error
+	// Read reads an archive from a Reader.
+	Read(input io.Reader, destination string) error
+	// Open extracts an archive file on disk while preserving uid/gid.
+	OpenPreserve(source, destination string) error
```

Comment:
I personally feel that extending this interface with more methods may be not required.  Having a few simple options that can be used to pass into Open() and Make() may do the trick.  Sort of something like:
```go
type Archiver interface {
  // ...
  Open(source, destination string, opts ...option)
  // ....
}

type option // ...
```